### DISCUSSION TITLE
NextAuth still redirects to localhost even with NEXTAUTH_URL defined

### DISCUSSION THREAD

**Discussion by dalner** [nextauthjs/next-auth#8879] [Help] [answered]
### Summary

I have spent many weeks trying to understand why an application with an authentication mechanism is still navigating to localhost even with very clear definitions of the actual domain its being hosted on defined in all variants of documentation one can find on the internet. Here's what I have: 

- Standalone deployment using docker using the [suggested documentation](https://github.com/vercel/next.js/tree/canary/examples/with-docker)
- Deployed into a Kubernetes cluster using an NGINX proxy 
- NEXTAUTH_URL, NEXT_PUBLIC_NEXTAUTH_URL both defined as build time args AND .env args that are loaded 
- middleware definition as suggested [here](https://next-auth.js.org/configuration/nextjs#example-default-value-1)

When I navigate to the exampleurl.com (which requires authentication) it will navigate to https://localhost:300/auth/signin?callbackUrl=%2f, if I modify that url to exampleurl.com/auth/signin?callbackUrl=%2f it will  navigate to the correct authentication flow without localhost redirect. I'm stumped here after trying dozens of different theories. I've read through all the variations of documentation (I'm on the latest nextjs and nextauth) and I cannot consider Vercel as a deployment strategy. Any help would be greatly appreciated, as I'm out of ideas. 

PS I've read through the middleware code that controls the redirect and my only suspicion is the nextauth url isn't being discovered in the process environments but it does read in the NEXT secret and all other variables defined. Stranger is the same build that is produced with the nextauth url defined in a docker compose deployment (in localhost) DOES redirect correctly using the nextauth url but only after the browser network loads the full javascript from localhost, so not sure if its how it loads in nextjs then nextauth?

**Comment by dalner** [comment, marked as answer]
Apparently a update was just recently posted in the nextjs community I missed. Bumped the version and it now works. :)

**Reply by ggorlen** [reply]
Could you share which version you bumped from, and which version you bumped to? I assume you bumped the [next-auth](https://www.npmjs.com/package/next-auth) package rather than Next itself, or another library.
